proto/ignite/services/plugin/grpc/v1/service.proto

```diff
@@ -44,24 +45,42 @@ message ManifestResponse {
 
 message ExecuteRequest {
   ExecutedCommand cmd = 1;
+  uint32 client_api = 2;
 }
 
 message ExecuteResponse {}
 
 message ExecuteHookPreRequest {
   ExecutedHook hook = 1;
+  uint32 client_api = 2;
 }
 
 message ExecuteHookPreResponse {}
 
 message ExecuteHookPostRequest {
   ExecutedHook hook = 1;
+  uint32 client_api = 2;
 }
 
 message ExecuteHookPostResponse {}
 
 message ExecuteHookCleanUpRequest {
   ExecutedHook hook = 1;
+  uint32 client_api = 2;
 }
 
 message ExecuteHookCleanUpResponse {}
+
+// ClientAPIService defines the interface that allows plugins to get chain app analysis info.
+service ClientAPIService {
+  // GetChainInfo returns basic chain info for the configured app
+  rpc GetChainInfo(GetChainInfoRequest) returns (GetChainInfoResponse);
+
+  // Dependencies returns the app dependencies.
+  //rpc Dependencies(DependenciesRequest) returns (DependenciesResponse);
```

Comment:
What's this?

Reply:
It shouldn't be there right now, I will remove it 👍 

Dependencies is a feature that will be implemented to allow plugins to get app dependencies. It will initially be used to implement the Ignite App that will integrate with Telescope. It was originally done in #3561 which was merged agains this PR but it was decided to implement it in a separate PR.